cli/src/torii/routing.rs

```diff
@@ -149,18 +155,28 @@ async fn handle_pending_transactions(
     queue: Arc<Queue>,
     sumeragi: SumeragiHandle,
     pagination: Pagination,
-) -> Result<Scale<Vec<SignedTransaction>>> {
+    fetch_size: FetchSize,
+) -> Result<Scale<Batched<Vec<SignedTransaction>>>> {
     let query_response = sumeragi.apply_wsv(|wsv| {
         queue
             .all_transactions(wsv)
             .map(Into::into)
             .paginate(pagination)
             .collect::<Vec<_>>()
-        // TODO:
-        //.batched(fetch_size)
     });
+    let fetch_size = fetch_size.fetch_size.map_or_else(
+        || {
+            if query_response.len() > 0 {
+                NonZeroU32::new(query_response.len().try_into().unwrap()).unwrap()
+            } else {
+                // no particular reason why 1
+                NonZeroU32::new(1).unwrap()
+            }
+        },
+        |size| size,
+    );
 
-    Ok(Scale(query_response))
+    Ok(Scale(query_response.batched(fetch_size)))
```

Comment:
the response doesn't seem to be presisted anywhere. The whole point of batching (server-side cursor) is to keep fetching results from the original query